common/lib/protocol-definitions/src/blob.ts

```diff
@@ -0,0 +1,107 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { FileMode, IBlob, IAttachment, ITree, TreeEntry } from "./storage";
+import { SummaryObject, SummaryType } from "./summary";
+
+// duplicated this of common-utils
+function unreachableCase(_: never, message = "Unreachable Case"): never {
```

Comment:
We try to avoid implementations in definitions packages (interfaces/types only), so this file needs to live somewhere else.  Maybe common-utils, or some other implementation package already used by both client and server?  I don't really like the utils packages, but if we can at least consolidate to that package (or one like it) would be good.